src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```
@@ -719,6 +741,47 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        BundlePublishDirectory
+
+    Bundle the _ResolvedFileToPublish* items one file in PublishDir
+    (except those marked ExcludeFromSingleFile)
+    ============================================================
+    -->
+  <Target Name="_ComputeFilesToBundle"
+        Condition="'$(PublishSingleFile)' == 'true'">
+
+    <ItemGroup>
+      <_FilesToBundle Include="@(_ResolvedFileToPublishPreserveNewest)" 
+                      Condition="'%(_ResolvedFileToPublishPreserveNewest.ExcludeFromSingleFile)' != 'true'"/>
+      <_FilesToBundle Include="@(_ResolvedFileToPublishAlways)" 
+                      Condition="'%(_ResolvedFileToPublishPreserveNewest.ExcludeFromSingleFile)' != 'true'"/>
+    </ItemGroup>
+
+  </Target>
+
+  <UsingTask TaskName="GenerateBundle" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="BundlePublishDirectory" 
+          Condition="'$(PublishSingleFile)' == 'true'"
+          DependsOnTargets="_ComputeFilesToBundle" 
+          Inputs="@(_FilesToBundle)"
+          Outputs="$(PublishDir)\$(AssemblyName)$(_NativeExecutableExtension)">
```

```suggestion
          Outputs="$(PublishDir)$(AssemblyName)$(_NativeExecutableExtension)">
```

We ensure trailing slash and you can see this is how PublishDir is used elsewhere. This is another common idiom in msbuild, it prevents accidental writing to root of disk if something is undefined. I once deleted my entire hard drive this way so I stick to this now :)

---

Funny story, a similar issue caused CoreFx build perf to tank once, the cause `$(SomeProperty)\**\*.extension`.  Someone renamed `SomeProperty` and CI passed...